src/main/java/gvs/ui/logic/app/AppViewModel.java

```diff
@@ -41,15 +52,58 @@
   private static final String PROMT_MESSAGE = "no active session";
   private static final Logger logger = LoggerFactory
       .getLogger(AppViewModel.class);
+  
 
-  // TODO: do we still need the persistor here? @mtrentini
+  // TODO: do we still need the persistor here?
   public AppViewModel(ApplicationModel appModel,
-      ApplicationController appController, IPersistor persistor) {
+      ApplicationController appController, IPersistor persistor,
+      BorderPane rootLayout) {
     this.appModel = appModel;
     this.appModel.addObserver(this);
     this.appController = appController;
     this.persistor = persistor;
     this.currentSessionName.set(PROMT_MESSAGE);
+    this.rootLayout = rootLayout;
+    sessionNames
+        .addListener((ListChangeListener.Change<? extends String> c) -> {
+          if (!sessionIsInitialized) {
+            initSessionLayout();
+          }
+          if (sessionNames.size() == 1) {
+            displaySession();
+          } else if (sessionNames.isEmpty()) {
+            hideSession();
+          }
+        });
```

Comment:
Damit der Konstruktor kompakt bleibt, würde ich den change Listener in eine Funktion auslagern und als Function Pointer in der Funktion sentzen. (`sessionNames.addListener(this::myNewFunction)`)